src/agent/pyth/rpc/get_product.rs

```diff
@@ -27,6 +29,8 @@ where
     }?;
 
     let account = params.account.parse::<solana_sdk::pubkey::Pubkey>()?;
+    tracing::Span::current().record("account", account.to_string());
```

Comment:
Nitpick: This is the right way to add additional fields in the current span, would you mind modifying the PR description to point out that these tracing changes modify the current span introduced by `instrument` and is therefore async safe because it handles maintaining the ref (see: https://docs.rs/tracing/latest/tracing/struct.Span.html#method.enter) - we should make sure to avoid tracing logs that are handled by instrument and/or avoid entering new spans manually as you did here so I think as this is the first PR doing this it's worth documenting.